### PR TITLE
CI: schedule an online check for all formulae

### PR DESCRIPTION
**Pull request by SMillerDev**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----
This should make it faster to spot bitrot in homebrew/core.

### CODE REVIEW

File: .github/workflows/scheduled.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        os: [ubuntu-22.04]
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@main
+
+      - name: Set up Homebrew
+        id: set-up-homebrew
+        uses: Homebrew/actions/setup-homebrew@master
+
+      - name: Check all formulae for missing upstreams etc
```
Comment (MikeMcQuaid) [inline .github/workflows/scheduled.yml:29]:
```suggestion
      - name: Check all formulae for missing upstreams etc.
```

File: .github/workflows/scheduled.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  group: scheduled
+  cancel-in-progress: true
+
+jobs:
+  audit_online:
+    if: startsWith( github.repository, 'Homebrew/' )
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-22.04]
```
Comment (MikeMcQuaid) [inline .github/workflows/scheduled.yml:20]:
```suggestion
    runs-on: ubuntu-22.04
```

File: .github/workflows/scheduled.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@main
+
```
Comment (MikeMcQuaid) [inline .github/workflows/scheduled.yml:24]:
```suggestion
```

`setup-homebrew` will do this

File: .github/workflows/scheduled.yml [hunk 4]
```diff
@@ -0,0 +1,30 @@
+name: Scheduled online check
+on:
+  push:
+    paths:
+      - .github/workflows/scheduled.yml
+  schedule:
+    # Once every day
```
Comment (MikeMcQuaid) [inline .github/workflows/scheduled.yml:7]:
```suggestion
    # Once every day at 1AM
```

**Review by MikeMcQuaid — DISMISSED** [review]

**Comment by fxcoudert** [comment]
- Where will the output go? How does it report errors?
- That does not include source and resources, right?

**Review by MikeMcQuaid — DISMISSED** [review]

**Comment by SMillerDev** [comment]
> Where will the output go? How does it report errors?

In this very minimal version, just the logs. If we move away from Google Analytics we could log it to influxdb though.

> That does not include source and resources, right?

No, I still want to find a way to run `brew fetch` for that.

**Review by chenrui333 — DISMISSED** [review]

**Comment by SMillerDev** [comment]
This works much nicer: https://github.com/SMillerDev/homebrew-core/actions/runs/3706904058 10 random formulae are now tested and their sources fetched every day.

**Comment by SMillerDev** [comment]
And now it found something: https://github.com/SMillerDev/homebrew-core/actions/runs/3706958400

**Comment by fxcoudert** [comment]
But that homepage is reachable for me https://scalasca.org/software/cube-4.x/download.html
Is there a retry implemented to prevent network flukes?

**Comment by fxcoudert** [comment]
> Doesn't seem like a very high price.

Both jobs that ran report 30 s of start-up time for a 3 s job.

> And python formulae can have a lot of dependencies.

Does `brew audit --online --skip-style --only github_repository_archived,gitlab_repository_archived,homepage` check all dependencies? I feel dumb but I don't understand, really.

**Comment by SMillerDev** [comment]
Damned, I meant to write resources there. Which would manifest in a much slower `brew fetch` step.

**Comment by fxcoudert** [comment]
Ah ok, so this is a first step towards a full fetch. Makes more sense, sorry.

**Comment by SMillerDev** [comment]
I started out without a good idea how to do a full fetch but I would say after yesterday this is already it. This has 3 components:

1. Pick 20 random formulae to test
1. Audit the homepage and repo state (archived etc)
1. Run `brew fetch -s` on the formula.

File: .github/workflows/scheduled.yml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    # Once every day at 1AM
+    - cron: "0 1 * * *"
+
+concurrency:
+  group: scheduled
+  cancel-in-progress: true
+
+jobs:
+  create_matrix:
+    # if: startsWith( github.repository, 'Homebrew/' )
```
Comment (SMillerDev) [inline .github/workflows/scheduled.yml:16]:
```suggestion
    if: startsWith( github.repository, 'Homebrew/' )
```

File: .github/workflows/scheduled.yml [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+      - name: Set up Homebrew
+        id: set-up-homebrew
+        uses: Homebrew/actions/setup-homebrew@master
+      - id: matrix
+        run: |
+          formula=$(find  $(brew --repo "${TAP}")/Formula |  shuf -n "${TEST_COUNT}" | xargs -I{} basename {} .rb)
+          json=$(brew info --json=v2 $formula | jq --compact-output '[.formulae[].name]')
+          echo "json=${json}" >> $GITHUB_OUTPUT
+  audit_online:
+    # if: startsWith( github.repository, 'Homebrew/' )
```
Comment (SMillerDev) [inline .github/workflows/scheduled.yml:33]:
```suggestion
    if: startsWith( github.repository, 'Homebrew/' )
```

**Review by Rylan12 — COMMENTED** [review]
This seems great 👍! Once the commented lines are un-commented I'd say let's give it a try

**Review by Rylan12 — APPROVED** [review]

**Pull request merged**